lib/puppet/provider/cs_location/crm.rb

```diff
@@ -19,6 +19,14 @@
 
   mk_resource_methods
 
+  # we need to check if we run at least pacemakerd version 1.1.13 before enabling feature discovery
+  # see http://blog.clusterlabs.org/blog/2014/feature-spotlight-controllable-resource-discovery
+  unless Facter.value('pacemakerd_version').nil?
+    if Puppet::Util::Package.versioncmp(Facter.value('pacemakerd_version'), '1.1.13') >= 0
+      has_feature :discovery
+    end
```

Comment:
since i don't know any way to force reevaluation of a fact, we could duplicate the version check from the fact here

Reply:
@igalic How? Can we just use the same code (Facter::Util::Resolution...) here? Will that work?

Reply:
> since i don't know any way to force reevaluation of a fact, we could duplicate the version check from the fact here

AFAIK by design you can't re-evaluate facts. I also think `has_feature` is evaluated during the catalog compilation, not while running it.

I'd suggest to expand the `pacemakerd_version` fact with some guess work.

You can query the package manager, but there's potential complications. For example, yum might do a network query which can mean a higher server load. Without it, it may not know which version would be installed. There may be a `yumrepo` resource within the catalog. A version parameter may be set. Similar concerns apply to other package managers.

Another is to use the OS name/version to make a guess.

I've never used corosync / pacemaker so I don't know what's appropriate but it's a known limitation of Puppet where you basically can't converge in a single run.

Reply:
@ekohl 
That sounds a bit horrible.
I would rather propose to go back to the solution without the custom fact and getting the version with the pacemakerd command in the provider (basically the state, which was committed by me here: https://github.com/voxpupuli/puppet-corosync/pull/450/commits/6ecabc13d978a3c85a878ca5f44ccd6eec7cbdb2 plus improvements).

This also would mean i need to go this way, which i already mentioned earlier:

> I would suggest to revert the idea with the pacemakerd_version fact and trying to let the previous implementation (gathering the pacemakerd version within the provider) passing the tests.
> However i don't know how to fix the acceptance tests then. As said previously redhat vs. non-redhat is definitly wrong. Would it be feasible to check the OS and the major version there and see, which one has pacemaker version >= 1.1.13 and which < 1.1.13, rather than trying to check the version itself directly? Would that be a valid way to do it?

What do you think?

Reply:
that's kinda what i suggested, when i said to duplicate the facter code in the provider, but if the fact isn't used anywhere else, you might as well only have it in the provider.

Reply:
It is rather horrible, I agree. I do think that guessing based on OS name/version is for a lot of software the most reasonable thing to do.

Reply:
I think getting the version in the provider is better and stable, than making guesswork in the custom fact.

I will go ahead and implement my proposal and tune the tests. In addition to that i will rebase against current master, since some time has passed by. 

Thanks so far for your inputs.